grid.py

```diff
@@ -23,139 +19,99 @@ def __init__(self, num_rows, num_cols, tile_length):
-    #             row = self._get_idx(y_obst, True)
-    #             if (self.grid[row][col].isObstacle == False):
-    #                 if row > len(self.grid) or col > len(self.grid[0]):
-    #                     # TODO handle offgrid case
-    #                     return
-    #                 if (self.grid[row][col] in path):
-    #                     returner = True
-    #                 self.grid[row][col].isObstacle = True
-    #                 if (self.bloat_tile(row, col, path) == True):
-    #                     returner = True
-    #
-    #     for i in range(len(sensorDataBot)):
-    #         angle = ir_mappings_bot[i]
-    #         distance = sensorDataBot[i]
-    #         if distance != -1:
-    #             x_obst = x + robot_radius + distance * math.cos(angle)
-    #             # y_obst = y + radius + distance * math.sin(angle)
-    #             y_obst = y + robot_radius - distance * \
-    #                      math.sin(angle)  # upper left origin
-    #             col = self._get_idx(x_obst, False)
-    #             row = self._get_idx(y_obst, True)
-    #             if (self.grid[row][col].isObstacle == False):
-    #                 if row > len(self.grid) or col > len(self.grid[0]):
-    #                     # TODO handle offgrid case
-    #                     return
-    #                 if (self.grid[row][col] in path):
-    #                     returner = True
-    #                 self.grid[row][col].isObstacle = True
-    #                 if (self.bloat_tile(row, col, robot_radius) == True):
-    #                     returner = True
-    #
-    #     for i in lidarData:
-    #         angle = i[0]
-    #         distance = i[1]
-    #         if distance != -1:
-    #             x_obst = x + robot_radius + distance * math.cos(angle)
-    #             # y_obst = y + radius + distance * math.sin(angle)
-    #             y_obst = y + robot_radius - distance * \
-    #                      math.sin(angle)  # upper left origin
-    #             col = self._get_idx(x_obst, False)
-    #             row = self._get_idx(y_obst, True)
-    #             if (self.grid[row][col].isObstacle == False):
-    #                 if row > len(self.grid) or col > len(self.grid[0]):
-    #                     # TODO handle offgrid case
-    #                     return
-    #                 if (self.grid[row][col] in path):
-    #                     returner = True
-    #                 self.grid[row][col].isObstacle = True
-    #                 if (self.bloat_tile(row, col, robot_radius) == True):
-    #                     returner = True
-    #         return returner
-
-    def updateGridLidar(self, x, y, lidarData, radius, bloat_factor, pathSet, fullGrid):
-        """updates the grid based on the lidarData passed in
+    def update_grid(self, x, y, sensor_state: SensorState, radius, bloat_factor, path_set = set()):
+        for sensor_type, sensor_data in enumerate(sensor_state.sensor_data):
```

Comment:
Add in [sensor_state.terabee_bot, sensor_state.terabee_mid ... ] so we can get rid of sensor_state.sensor_data attribute

Reply:
resolved